File: Study_Python/Programmers_시저암호.py
'''
TIL

import string 후
list(string.ascii_lowercase)를 사용하면 ['a', 'b', 'c', ...] 과 같은
소문자 list를 생성할 수 있다.

list이름.index(값) => 해당 값이 있는 index를 알려준다
'''

import string

def solution(s, n):
    answer = ''
    
    small = list(string.ascii_lowercase)
    large = list(string.ascii_uppercase)
    
    for i in range(len(s)):
        if s[i] == ' ':
            answer += " "
            continue
        
        if s[i].islower():
            answer += small[(small.index(s[i]) + n) % 26]
        else:
            answer += large[(large.index(s[i]) + n) % 26]
            
    return answer

File: Programmers/Programmers_시저암호.java
import java.util.*;

class Solution {
    public String solution(String s, int n) {
        String answer = "";
        char[] small = new char[26];
        char[] large = new char[26];
        
        char smallA = 'a';
        char largeA = 'A';
        
        for(int i=0; i<small.length; i++){
            small[i] = (char)(smallA+i);
            large[i] = (char)(largeA+i);
        }//초기화
        
        for(int i=0; i<s.length(); i++){
            char c = s.charAt(i);
            if(c == ' '){
                answer += " "; 
                continue;
            }
            
            if(c >= 'a' && c <= 'z'){
                int index = c - 'a';
                answer += small[(index + n) % 26] + "";
            }else{
                int index = c - 'A';
                answer += large[(index + n) % 26] + "";
            }
        }
        
        return answer;
    }
}